frappe/erpnext
# refactor: update_serial_no function for old Maintenance Visits

### Issue

- Old documents which were saved before the refactor would not have the "Maintenance Schedule Detail" field set.

![image](https://user-images.githubusercontent.com/43572428/145804033-e6b9371b-e1e6-484f-9efd-20b551954a08.png)

### Fix
- Using `maintenance_schedule` field in **"Maintenance Visit"** to fetch the serial nos instead of the **Maintenance Schedule Detail** row id, which isn't possible to identify on old docs.
- Patch to update this field on old docs that have `prevdoc_docname`.

**To Test:**
- Make an independent maintenance visit (not from maintenance schedule) and see if it works fine. Use serialised item in it and add serial nos > Add multiple serialised items and see if you can select appropriate serial nos (check the filters)
- Make a maintenance schedule with multiple items, 1 non serialised, 2 serialised. Make visits, see if it works and updates the schedule correctly
